docs/source/developers/bug_reports.rst

```diff
@@ -203,59 +190,24 @@ Identify affected version
 +++++++++++++++++++++++++
 
 If you're reporting something that used to work in a previous version
-but doesn't work in the current release, you can add the **Affects version**
-field to identify the earliest known version where the bug is observed.
-For feature requests and other proposals, leave **Affects version** empty as
-it is not applicable.
+but doesn't work in the current release, you add the **Affects version**
+in the description to identify the earliest known version where the bug is
+observed.
 
 .. _issue-lifecycle:
 
 Issue lifecycle
 ===============
 
-Both bug reports and feature requests follow a defined lifecycle. The issue
-**Status** field is used to document the current state of the issue, while the
-**Resolution** field indicates the outcome of issues that have reached
-terminal status. 
-
-
-.. _issue-status:
-
-Issue Status
-++++++++++++
-
-The Arrow project uses the following statuses in Jira to indicate what has -
-and will be - done on an issue:
-
-* **Open** - This is the initial issue state, prior to a contributor assigning
-  the issue and starting progress. Issues in this state should be unassigned.
-* **In progress** - At the time a contributor self-assigns an issue, the status
-  should be set to In progress by clicking the **Start progress** button. All
-  issues in this status should have an assignee - unassigned issues will be
-  set back to a status of Open. Issues remain "in progress" until resolved or
-  closed, including during review of pull requests.
-* **Resolved** - This is a terminal status indicating action has been taken
-  on the issue, which is now considered completed. Issues in a resolved status
-  should have a resolution code set to **Fixed**.
-* **Closed** - Another terminal status, Closed indicates the issue is complete,
-  but *without* action being taken. The following resolution codes apply to
-  issues in Closed status:
-
-  * Won't Fix
-  * Duplicate
-  * Invalid
-  * Incomplete
-  * Cannot Reproduce
-  * Not a Problem
-  * Not a Bug
-  * Workaround
-  * Information Provided
-  * Works for Me
-  * Won't Do
-  * Abandoned
-
-* **Reopened** - When an issue has been closed or resolved, but additional
-  attention is needed, it may be reopened.
+Both bug reports and feature requests follow a defined lifecycle. If the issue
+is currently worked on it has a developer assigned. When the issue has reached
+a terminal status, it is labeled as closed.
+
+* **Closed as completed** - indicates the issue is complete and the PR that
+  resolved the issue should be added in a following comment.
+
+* **closed as not planned** - indicates the issue is complete,
+  but *without* action being taken.
```

Comment:
Here is a suggestion to rephrase this (note that GitHub auto-links PRs to their corresponding issues, so I'm not sure you need to add an issue comment to mention a PR?):
```suggestion
Both bug reports and feature requests follow a defined lifecycle. If an issue
is currently worked on, it should have a developer assigned. When an issue has
reached a terminal status, it is closed with one of two outcomes:

* **Closed as completed** - indicates the issue is complete; the PR that
  resolved the issue should have been automatically linked by GitHub
  (assuming the PR correctly mentioned the issue number).

* **closed as not planned** - indicates the issue is closed and should
  not receive any further updates, but *without* action being taken.
```

Reply:
> note that GitHub auto-links PRs to their corresponding issues, so I'm not sure you need to add an issue comment to mention a PR?

We indeed don't need to mention that, since the automation of the script will add this comment to say the issue was closed (so you don't have to do this manually). But personally I do think such comments are useful, since the autolinking by GitHub itself doesn't generate notifications.

Reply:
> But personally I do think such comments are useful, since the autolinking by GitHub itself doesn't generate notifications.

Just to be clear, do you manually add a comment ([example](https://github.com/apache/arrow/issues/14854#issuecomment-1342535500)) when closing an issue? If yes, then I would add this in the docs also.

Reply:
Added proposed changes and extra sentence about why it would be nice to add a comment that the issue was closed:
https://github.com/apache/arrow/pull/14919/commits/837c7d98e91259981c510d7e165e61248ce75e83